packages/flutter_test/test/controller_test.dart

```diff
@@ -935,6 +935,445 @@ void main() {
           containsAllInOrder(expectedMatchers));
       });
     });
+
+    group('actions', () {
+      testWidgets('performAction with unsupported action throws StateError', (WidgetTester tester) async {
+        await tester.pumpWidget(Semantics());
+
+        expect(() => tester.semantics.performAction(tester.semantics.find(find.byType(Semantics)), SemanticsAction.tap), throwsStateError);
+      });
+
+      testWidgets('tap causes semantic tap', (WidgetTester tester) async {
+        bool invoked = false;
+        debugDumpSemanticsTree();
+        await tester.pumpWidget(
+          MaterialApp(
+            home: TextButton(
+              onPressed: () => invoked = true,
+              child: const Text('Test Button'),
+            ),
+          ),
+        );
+
+        tester.semantics.tap(find.byType(TextButton));
+        expect(invoked, isTrue);
+      });
+
+      testWidgets('longPress causes semantic long press', (WidgetTester tester) async {
+        bool invoked = false;
+        await tester.pumpWidget(
+          MaterialApp(
+            home: TextButton(
+              onPressed: () {},
+              onLongPress: () => invoked = true,
+              child: const Text('Test Button'),
+            ),
+          ),
+        );
+
+        tester.semantics.longPress(find.byType(TextButton));
+        expect(invoked, isTrue);
+      });
+
+      testWidgets('scrollLeft and scrollRight scroll left and right respectively', (WidgetTester tester) async {
+        await tester.pumpWidget(MaterialApp(
+          home: ListView(
+            scrollDirection: Axis.horizontal,
+            children: <Widget>[
+              SizedBox(
+                height: 40,
+                width: tester.binding.window.physicalSize.width * 1.5,
+              )
+            ],
+          ),
+        ));
+
+        expect(
+          tester.semantics.findScrollable(find.byType(ListView)),
+          containsSemantics(hasScrollLeftAction: true, hasScrollRightAction: false),
+          reason: 'When not yet scrolled, a scrollview should only be able to support left scrolls.');
```

Comment:
formatting nit: place `);` on the next line, here and elsewhere

Reply:
Done!